modules/imgcodecs/test/test_png.cpp

```diff
@@ -7,6 +7,19 @@ namespace opencv_test { namespace {
 
 #if defined(HAVE_PNG) || defined(HAVE_SPNG)
 
+TEST(imRead, passing_Mat)
+{
+  Mat img = Mat::zeros(1000, 1000, CV_8UC3);
+  Mat subImg = img.rowRange(0, 512).colRange(0, 512);
+  imread("../cv/shared/lena.png", subImg);
+  Mat ori = imread("../cv/shared/lena.png");
+  Mat diff;
+  absdiff(ori, subImg, diff);
+  EXPECT_EQ(0, sum(diff)[0]);
```

Comment:
Replace to `EXPECT_EQ(cv::norm(ori, subImg, NORM_INF), 0)`

Reply:
done